src/documents/views.py

```diff
@@ -729,6 +731,65 @@ def share_links(self, request, pk=None):
             ]
             return Response(links)
 
+    @action(methods=["get"], detail=True, name="Audit Trail")
+    def history(self, request, pk=None):
+        if not settings.AUDIT_LOG_ENABLED:
+            return HttpResponseBadRequest("Audit log is disabled")
+        try:
+            doc = Document.objects.get(pk=pk)
+            if not request.user.has_perm("auditlog.view_logentry") or (
+                doc.owner is not None and doc.owner != request.user
+            ):
+                return HttpResponseForbidden(
+                    "Insufficient permissions",
+                )
+        except Document.DoesNotExist:  # pragma: no cover
+            raise Http404
+
+        if request.method == "GET":
```

Comment:
Does `@action` already validate the method?

Reply:
Ah, yea I think that was leftover from before I discovered the kwarg. Confirmed you get a 405 without it, thanks.